hash_practice/exercises.py

```diff
@@ -2,18 +2,33 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    hash_map = {}
+    for string in strings:
+        sorted_string = "".join(sorted(string))
+        if sorted_string in hash_map:
+            hash_map[sorted_string].append(string)
+        else:
+            hash_map[sorted_string] = [string]
+    return list(hash_map.values())
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
Don't forget to account for the O(k * log(k)) sorting you perform in line 30, where k is the number of unique numbers in the list. As k trends to n if every number is unique, I would consider this O(n * log(n))